packages/flutter_tools/lib/src/build_system/targets/localizations.dart

```diff
@@ -0,0 +1,285 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+import 'package:yaml/yaml.dart';
+
+import '../../artifacts.dart';
+import '../../base/file_system.dart';
+import '../../base/io.dart';
+import '../../base/logger.dart';
+import '../../globals.dart' as globals;
+import '../build_system.dart';
+import '../depfile.dart';
+
+/// Run the localizations generation script with the configuration [options].
+Future<Depfile> generateLocalizations({
+  @required LocalizationOptions options,
+  @required String flutterRoot,
+  @required FileSystem fileSystem,
+  @required ProcessManager processManager,
+  @required Logger logger,
+  @required Directory projectDir,
+  @required String dartBinaryPath,
+}) async {
+  // Setup project inputs and outputs. This currently makes some
+  // guess about the output and does not include all files.
+  final Directory inputArb = fileSystem.directory(
+    options.arbDirectory ?? projectDir
+      .childDirectory('lib')
+      .childFile('l10n').uri,
+  );
+
+  File outputLocalzations;
+  if (options.outputLocalizationsFile != null) {
+    outputLocalzations = fileSystem
+      .file(inputArb.uri.resolveUri(options.outputLocalizationsFile));
+  } else {
+    outputLocalzations = projectDir
+    .childDirectory('lib')
+    .childFile('app_localizations.dart');
+  }
+
+  if (!inputArb.existsSync()) {
+    logger.printError('arb-dir: ${inputArb.path} does not exist.');
+    throw Exception();
+  }
+
+  final List<File> inputs = <File>[
+    // Include all arb files as build inputs.
+    for (final File file in inputArb.listSync().whereType<File>())
+      if (fileSystem.path.extension(file.path) == '.arb')
+        file,
+  ];
+  final List<File> outputs = <File>[
+    outputLocalzations,
+  ];
+
+  final Depfile depfile = Depfile(inputs, outputs);
+  final String genL10nPath = fileSystem.path.join(
+    flutterRoot,
+    'dev',
+    'tools',
+    'localization',
+    'bin',
+    'gen_l10n.dart',
```

Comment:
Does this make it into the inputs somehow as well?

Reply:
No. since we're running it from source it wouldn't be sufficient to capture the dependencies either.

Reply:
Please leave a comment here about that with a link to a tracking issue.

Reply:
Added a comment below on the Target inputs